pkg/api/controllers/catalog/catalog.go

```diff
@@ -142,3 +224,47 @@ func cleanupPath(dir string, files []string, valid map[string]bool) int {
 	}
 	return count
 }
+
+func checkCatalogUpdate(m *manager.Manager, catalog *v3.Catalog, cmt *manager.CatalogInfo) (runtime.Object, error) {
+	hashPath := helmlib.CatalogSHA256Hash(catalog)
+	path := filepath.Join(helmlib.CatalogCache, hashPath)
+	if _, err := os.Stat(path); err == nil {
+		if git.IsValid(catalog.Spec.URL) {
```

Comment:
Instead of checking git.IsValid we could probably check for the existence of a hashPath/.git directory and early exit if not exists.

Reply:
updated.